packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -315,7 +320,8 @@ export async function buildExtraViteConfig(
 	const extraViteConfig: Partial<UserConfig> = {
 		resolve: {
 			mainFields: [...SVELTE_RESOLVE_MAIN_FIELDS],
-			dedupe: [...SVELTE_IMPORTS, ...SVELTE_HMR_IMPORTS]
+			dedupe: [...SVELTE_IMPORTS, ...SVELTE_HMR_IMPORTS],
+			conditions: [...SVELTE_EXPORT_CONDITIONS]
```

Comment:
we only add these conditions and let vite merge them. if there are other conditions already, svelte might have a lower priority. should we unshift them to try and ensure they are first?

Reply:
Yeah, I think svelte should come first

Reply:
actually it looks like vite does not expose the default resolve extensions config, so the values we see here would all be custom conditions added by users or other plugins. i don't think we need to try and stay first of the custom ones.

Given that the testcase works i'm inclined to leave it that way until someone comes up with a strange nested condition need. @bluwy how does this work in astro if you use multiple frameworks at the same time that all require custom extensions?

Reply:
Oh yeah [looks like so](https://github.com/builderio/vite/blob/b80f7c744f3b14e725799c2a4ef69af97d738216/packages/vite/src/node/config.ts#L485). So we could leave it as is 👍 



> @bluwy how does this work in astro if you use multiple frameworks at the same time that all require custom extensions?

I don't think it'll be a problem as it doesn't make sense to do `pkg.exports['./'].astro.svelte` or `pkg.exports['./'].svelte.solid`.

Reply:
thanks. so we'll leave it as is for now